cartoframes/context.py

```diff
@@ -247,14 +249,29 @@ def read(self, table_name, limit=None, index='cartodb_id',
 
         query = 'COPY {source} TO stdout WITH (FORMAT csv, HEADER true)'.format(source=source)
 
-        result = self.copyClient.copyto_stream(query)
+        result = self._recursiveRead(query, retry_times)
         df = pd.read_csv(result)
 
         if decode_geom:
             df['geometry'] = df.the_geom.apply(_decode_geom)
 
         return df
 
+    def _recursiveRead(self, query, retry_times):
+        try:
+            return self.copyClient.copyto_stream(query)
+        except CartoRateLimitException as err:
+            retry_times -= 1
+            if retry_times > 0:
+                print('Read call rate limited. Waiting {s} seconds'.format(s=err.retryAfter))
```

Comment:
Maybe there's a better option, but I'd stick to the `warnings` module for this (or maybe we should configure a proper logger).

Reply:
The `warnings.warn` leave too much information for the user, not only the message.

I will change it and :+1: to `we should configure a proper logger`

Reply:
You can close an issue if you add a logger :) https://github.com/CartoDB/cartoframes/issues/422